openhands/runtime/impl/action_execution/action_execution_client.py

```diff
@@ -217,7 +217,7 @@ def send_action_for_execution(self, action: Action) -> Observation:
 
         # set timeout to default if not set
         if action.timeout is None:
-            action.timeout = self.config.sandbox.timeout
+            action.set_default_timeout(self.config.sandbox.timeout)
```

Comment:
can we call this `soft timeout`?

Reply:
this is not quite soft time, but rather "set hard timeout without blocking", maybe I can just add an additional argument (blocking=False) to "set_hard_timeout" function

Reply:
done most recent commit